### DISCUSSION TITLE
How to choose compression method for lkrm images?

### DISCUSSION THREAD

**Discussion by grivalera** [ipxe/ipxe#1240] [Ideas]
How to choose compression method for lkrm images?
I tried:
`file ipxe.lkrn`

Its shows me:
`ipxe.lkrn: Linux kernel x86 boot executable bzImage, version 1.0.0+, RO-rootFS,`

As I got, now it is bzip for ipxe code, but for old machines its to slow to decompress bzip. Today linux kernels gives at least four methods for compress kernel, lzop, for example very fast for decompress. How I can change compression way?

**Comment by mcb30** [comment]
There is no way to change the compression algorithm (other than by disabling it completely).

The size of the iPXE binary is so small that I doubt you'll be able to measure the decompression time on any meaningful human scale.

**Reply by grivalera** [reply]
it boots for a half of second, may be less, but depacking time at the same 486 seems like ipxe "broken", because I should wait for 10 seconds, maybe. There is no any kind of progress bar, so still wait while it depacking or completely frozen.

**Reply by grivalera** [reply]
It hasn't sense for any machines, except very old.

**Reply by grivalera** [reply]
I've checked it just now, Pentium 200mmx  decompress it about 6 seconds. Uff...

**Reply by mcb30** [reply]
> I've checked it just now, Pentium 200mmx decompress it about 6 seconds. Uff...

Nope.  Decompressing the 330kB all-drivers image (`bin/ipxe.dsk`) uses around 130M instructions.  Decompressing a 69kB single-driver image (`bin/intel.rom`) uses around 27M instructions.

To get even the larger of those binaries to take 6 seconds to decompress, you'd need to have a CPU running at around 20 MIPS, which is around half the speed that you'd get from a typical 486.  Whatever you're measuring on a Pentium 200MMX, it's not the decompression time.

**Reply by grivalera** [reply]
It was full pack (ipxe.lkrn), agree, special driver get less time to depack... It is my first line to boot - versatile ipxe - second line - rtl8139 ipxe :-)

**Reply by grivalera** [reply]
But I can proove, its long time to decompress on pentium 200mmx, and very long at 486.

**Reply by mcb30** [reply]
> But I can proove, its long time to decompress on pentium 200mmx, and very long at 486.

You're timing something else. What makes you think you've identified the time spent decompressing?
